### PR TITLE
Add sequence classification support for small Gemma 3 text models

### LINKED ISSUE #40554: Add sequence classification support for small Gemma 3 models
### Feature request

While #39465 added support for larger Gemma 3 (4b parameters or more) via `Gemma3ForSequenceClassification`, it did not add a `Gemma3TextForSequenceClassification` as for example proposed in https://github.com/huggingface/transformers/issues/36755#issuecomment-2929098303. However this is necessary to use the small model variants with 1b and 270m parameters.

### Motivation

Especially the smallest model with 270m parameters might be an interesting replacement for older model architectures like DistilBERT in resource-constrained environments. We have seen reasonable performance in full-parameter conversation fine-tuning, but being able to use a task-specific head for sequence classification would improve things considerably.

### Your contribution

We could test proposed changes by adapting our existing training scripts which currently only work with 4b parameters or larger model variants.

### PR DESCRIPTION
**Pull request by abdokaseb**
# What does this PR do?

Add Gemma3TextForSequenceClassification for Gemma3 small text models following generic practice

Testing:
- Tested with single-node option, loss curve and AUC are all expected with no issue.
- All testes passed
- make fixup passed

Fixes #40554 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@ArthurZucker

### CODE REVIEW

**Review by zucchini-nlp — APPROVED** [review]
Hey @abdokaseb , thanks for the PR! Indeed, models saved without an image tower can't be loaded with existing class, totally forgot about them

**Comment by zucchini-nlp** [comment]
Failing tests are not related, a rebase might help. I will merge a PR to fix processor tests in an hour or so

**Comment by zucchini-nlp** [comment]
@abdokaseb, the workers are crashing and thus we can't merge the PR yet. I will check what is wrong and merge when fixed

**Comment by zucchini-nlp** [comment]
The CI is very flaky lately 😿

**Comment by adamreichold** [comment]
> The CI is very flaky lately 😿

@zucchini-nlp @abdokaseb Thank you for carrying it over the finish line nevertheless!

**Comment by ArthurZucker** [comment]
🤗
